tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -315,10 +315,28 @@ public function filterResponseByNotificationSubjectAndResource(string $subject,
 	 * @param TableNode $table
 	 *
 	 * @return void
+	 * @throws Exception
 	 */
 	public function userShouldGetANotificationWithMessage(string $user, string $subject, TableNode $table):void {
+		$count = 0;
 		$this->userListAllNotifications($user);
-		$actualMessage = str_replace(["\r", "\n"], " ", $this->filterResponseAccordingToNotificationSubject($subject)->message);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200);
+		// sometimes the test might try to get notification before the notification is created by the server
+		// in order to prevent test from failing because of that try to list the notifications again
+		while (!isset($this->filterResponseAccordingToNotificationSubject($subject)->message) && $count <= 5) {
+			if ($count > 0) {
+				\sleep(1);
+			}
```

Comment:
```suggestion
			\sleep(1);
```
since we have attempted already one time in up ?

Reply:
right!
or remove:
```
$this->userListAllNotifications($user);
$this->featureContext->theHTTPStatusCodeShouldBe(200);
```

Reply:
This is needed 
```php
$this->userListAllNotifications($user);
$this->featureContext->theHTTPStatusCodeShouldBe(200);
```
I'll remove the sleep

Reply:
Or 'do...while'?